ambry-store/src/test/java/com.github.ambry.store/IndexSegmentTest.java

```diff
@@ -454,17 +512,23 @@ private void doGetEntriesSinceTest(NavigableMap<MockId, IndexValue> referenceInd
    * @param startOffset the expected start {@link Offset} of the {@link IndexSegment}
    * @param numItems the expected numner of items the {@code indexSegment}
    * @param endOffset the expected end offset of the {@code indexSegment}
+   * @param lastModifiedTimeInSecs the last modified time of the index segment in secs
+   * @param resetKey the resetKey of the index segment
    * @param extraOffsetsToCheck a {@link Map} that defines the put record offsets of keys whose presence needs to be
    *                            verified in the {@link Journal}.
    * @throws IOException
    * @throws StoreException
    */
   private void verifyReadFromFile(NavigableMap<MockId, IndexValue> referenceIndex, File file, Offset startOffset,
-      int numItems, long endOffset, Map<Offset, MockId> extraOffsetsToCheck) throws IOException, StoreException {
+      int numItems, long endOffset, long lastModifiedTimeInSecs,
+      Pair<StoreKey, PersistentIndex.IndexEntryType> resetKey, Map<Offset, MockId> extraOffsetsToCheck)
+      throws IOException, StoreException {
     // read from file (unmapped) and verify that everything is ok
     Journal journal = new Journal(tempDir.getAbsolutePath(), Integer.MAX_VALUE, Integer.MAX_VALUE);
     IndexSegment fromDisk = createIndexSegmentFromFile(file, false, journal);
-    verifyIndexSegmentDetails(fromDisk, startOffset, numItems, false, endOffset);
+    lastModifiedTimeInSecs =
```

Comment:
once again, I think the caller should have sent this correctly. This function is just verifying the values that were sent.